flixel/addons/display/FlxPieDial.hx

```diff
@@ -0,0 +1,260 @@
+package flixel.addons.display;
+
+import flixel.FlxSprite;
+import flixel.math.FlxPoint;
+import flixel.math.FlxVector;
+import flixel.util.FlxColor;
+import flixel.util.FlxSpriteUtil;
+import haxe.EnumTools;
+import openfl.display.BitmapData;
+import openfl.display.BitmapDataChannel;
+import openfl.display.BlendMode;
+
+/**
+ * A dynamic shape that fills up the way a pie chart does. Useful for timers and other things.
+ * @author larsiusprime
+ */
+class FlxPieDial extends FlxSprite
+{
+
+	public var amount(default, set):Float = 0;
+	private var pieFrames:Int = 0;
+	
+	public static inline var CIRCLE:String = "circle";
+	public static inline var SQUARE:String = "square";
+	
+	public function new(X:Float, Y:Float, Radius:Int, Color:FlxColor=FlxColor.WHITE, Frames:Int=36, Shape:String = CIRCLE, Clockwise:Bool = true, InnerRadius:Int=0) 
```

Comment:
The Constructor has `X` and `Y`, but they're ignored - instead `0, 0` is passed to `super()`.